omod/src/main/java/org/openmrs/module/mksreports/web/controller/MKSReportsManageController.java

```diff
@@ -13,196 +13,75 @@
-import org.apache.fop.apps.Fop;
-import org.apache.fop.apps.FopFactory;
-import org.apache.fop.apps.MimeConstants;
-import org.openmrs.Encounter;
 import org.openmrs.api.EncounterService;
 import org.openmrs.module.mksreports.MKSReportsConstants;
-import org.openmrs.module.mksreports.reports.PatientHistoryReportManager;
-import org.openmrs.module.patientsummary.PatientSummaryResult;
-import org.openmrs.module.patientsummary.PatientSummaryTemplate;
-import org.openmrs.module.patientsummary.api.PatientSummaryService;
-import org.openmrs.module.reporting.evaluation.context.EncounterEvaluationContext;
-import org.openmrs.module.reporting.query.encounter.EncounterIdSet;
-import org.openmrs.module.reporting.report.ReportDesign;
-import org.openmrs.module.reporting.report.service.ReportService;
-import org.openmrs.util.OpenmrsClassLoader;
+import org.openmrs.module.mksreports.reports.PatientHistoryPdfReport;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.ModelMap;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 
-/**
- * The main controller.
- */
 @Controller
 public class MKSReportsManageController {
 	
-	@Autowired
-	private EncounterService encounterService;
+	private PatientHistoryPdfReport pdfReport;
 	
-	/**
-	 * The path to the style sheet for Patient History reports.
-	 */
-	private static final String PATIENT_HISTORY_XSL_PATH = "patientHistoryFopStylesheet.xsl";
-	
-	@Autowired
-	private PatientSummaryService patientSummaryService;
+	private EncounterService es;
 	
 	@Autowired
-	private ReportService reportService;
+	public MKSReportsManageController(@Qualifier("encounterService") EncounterService es,
+	    PatientHistoryPdfReport pdfReport) {
+		this.es = es;
+		this.pdfReport = pdfReport;
+	}
 	
 	/**
 	 * Receives requests to run a patient summary.
 	 * 
 	 * @param patientId the id of patient whose summary you wish to view
-	 * @param summaryId the id of the patientsummary you wish to view
-	 * @param encounterId the id(s) of the encounters you wish to view, multiple encounters should be in
-	 *            the form "<uuid>,<uuid>"
+	 * @param encounterUuids the UUID(s) of the encounters you wish to view, CSV format is supported for
+	 *            multiple encounters
 	 * @param target if used as an href URL, the intended target of the <a> (e.g. _self, _blank)
 	 * @throws Exception
 	 */
 	@RequestMapping(value = MKSReportsConstants.CONTROLLER_PATIENTHISTORY_ROUTE)
 	public void renderPatientHistory(ModelMap model, HttpServletRequest request, HttpServletResponse response,
-	        @RequestParam("patientId") Integer patientId,
-	        @RequestParam(value = "encounterUuid", required = false) String encounterUuidParam,
+	        @RequestParam(value = "patientId", required = false) Integer patientId,
```

Comment:
This should be named `patientUuid` and be a way to convey the patient UUID.

Reply:
I believe it's actually expecting the patientId (given that PatientSummary is calling Cohort.addMember(Integer patientId) with that string.

Reply:
Do we want to change this to Uuid and lookup the id in the controller, or change signature of the new PatientHistoryPdfReport to require the Uuid and do the id lookup there before calling the patientsummary evaluate?

Reply:
@kml27 the controller will translate the patient UUID into whatever the downstream Java API needs. `PatientHistoryPdfReport` should either take POJOs as arguments or should take immediately ready/useful arguments. So let's stick to the latter for now (patient ID + encounter IDs).